apps/asap-server/test/controllers/events.test.ts

```diff
@@ -390,6 +390,95 @@ describe('Event controller', () => {
+      const lockedEventResponse = eventBase;
+      lockedEventResponse.flatData!.notesLocked = true;
+      lockedEventResponse.flatData!.videoRecordingLocked = true;
+      lockedEventResponse.flatData!.presentationLocked = true;
+      lockedEventResponse.flatData!.meetingMaterialsLocked = true;
+
+      nock(config.baseUrl)
+        .post(`/api/content/${config.appName}/graphql`, {
+          query: buildGraphQLQueryFetchEvent(eventId),
+        })
+        .reply(200, { data: { findEventsContent: lockedEventResponse } });
+
+      const result = await events.fetchById(eventId);
+      expect(result).toEqual({
+        ...eventResponse,
+        notes: null,
+        videoRecording: null,
+        presentation: null,
+        meetingMaterials: null,
+      });
+    });
+
+    test('Should return locked details if it enough time as passed and details are empty', async () => {
+      const emptyEvent = eventBase;
+      emptyEvent.flatData!.notes = null;
+      emptyEvent.flatData!.videoRecording = null;
+      emptyEvent.flatData!.presentation = null;
+      emptyEvent.flatData!.meetingMaterials = null;
+      emptyEvent.flatData!.notesLocked = null;
+      emptyEvent.flatData!.videoRecordingLocked = null;
+      emptyEvent.flatData!.presentationLocked = null;
+      emptyEvent.flatData!.meetingMaterialsLocked = null;
+
+      nock(config.baseUrl)
+        .post(`/api/content/${config.appName}/graphql`, {
+          query: buildGraphQLQueryFetchEvent(eventId),
+        })
+        .reply(200, { data: { findEventsContent: emptyEvent } });
+
+      const result = await events.fetchById(eventId);
+      expect(result).toEqual({
+        ...eventResponse,
+        notes: null,
+        videoRecording: null,
+        presentation: null,
+        meetingMaterials: null,
+      });
+    });
+
+    test('Should return empty (undefined) details if empty but the event is fresh', async () => {
+      const now = new Date().toISOString();
+      const emptyEvent = eventBase;
+      emptyEvent.flatData!.endDate = now;
+      emptyEvent.flatData!.notes = null;
+      emptyEvent.flatData!.videoRecording = null;
+      emptyEvent.flatData!.presentation = null;
+      emptyEvent.flatData!.meetingMaterials = null;
+      emptyEvent.flatData!.notesLocked = null;
+      emptyEvent.flatData!.videoRecordingLocked = null;
+      emptyEvent.flatData!.presentationLocked = null;
+      emptyEvent.flatData!.meetingMaterialsLocked = null;
+
+      nock(config.baseUrl)
+        .post(`/api/content/${config.appName}/graphql`, {
+          query: buildGraphQLQueryFetchEvent(eventId),
+        })
+        .reply(200, { data: { findEventsContent: emptyEvent } });
+
+      const result = await events.fetchById(eventId);
+      expect(result).toEqual({
+        ...eventResponse,
+        endDate: now,
+        notes: undefined,
+        videoRecording: undefined,
+        presentation: undefined,
+        meetingMaterials: [],
+      });
+    });
+  });
+
```

Comment:
You seem to be missing a test for when a value is actually returned (not `locked`, not stale, not undefined, just present) - unless it's already somewhere else?

Reply:
You could also test what happens when the `locked` flags or the materials themselves are not present in the flatData, as I believe will be the case for older events before we added these fields?

Reply:
The first one should be covered, since the coverage is 100% for the entity. I will see if I need to make the assertion explicit